src/state.rs

```diff
@@ -63,6 +66,8 @@ pub fn next_offer_id(store: &mut dyn Storage) -> StdResult<u8> {
     Ok(id)
 }
 
+// TODO: this could probably be just a Map
+
 pub const OFFER_NAMESPACE: &str = "offers";
 pub struct OfferIndexes<'a> {
     pub id: UniqueIndex<'a, u8, Offer>,
```

Comment:
seems like here would be useful to have 2 additional indexes 
by_sender
by_peer


usage of the following queries `query_offers_by_sender` and `query_offers_by_peer` is making the execute methods inefficient and you wont be able to scale without gas issues .

Reply:
This will also help your front end to query inbox/sent offers.